specifyweb/frontend/js_src/lib/components/Statistics/StatItems.tsx

```diff
@@ -0,0 +1,255 @@
+      isDefault={item.type === 'DefaultStat'}
+      query={query}
+      statLabel={statsSpecCalculated?.label}
+      statValue={item.itemValue}
+      onClick={handleClick}
+      onItemRename={handleItemRename}
+      onRemove={handleRemove}
+      onSpecChanged={
+        handleSpecChanged !== undefined
+          ? (tableName, fields) => {
+              handleSpecChanged(tableName, fields, statsSpecCalculated?.label);
+            }
+          : undefined
+      }
+      fields={statsSpecCalculated.fields}
+      tableName={statsSpecCalculated.tableName}
+      onItemValueLoad={handleItemValueLoad}
+    />
+  ) : item.type === 'DefaultStat' &&
+    statsSpecCalculated !== undefined &&
+    statsSpecCalculated.type === 'BackEndStat' &&
+    statsSpecCalculated.pathToValue !== undefined ? (
+    <BackEndItem
+      isDefault
+      urlToFetch={statsSpecCalculated.urlToFetch}
+      pathToValue={statsSpecCalculated.pathToValue}
+      statLabel={item.itemLabel}
+      statValue={item.itemValue}
+      onClick={handleClick}
+      onItemRename={handleItemRename}
+      onRemove={handleRemove}
+      formatter={statsSpecCalculated.formatter}
+      onItemValueLoad={handleItemValueLoad}
+    />
+  ) : null;
+}
+
+function BackEndItem({
+  statValue,
+  urlToFetch,
+  pathToValue,
+  formatter,
+  statLabel,
+  isDefault,
+  onClick: handleClick,
+  onRemove: handleRemove,
+  onItemRename: handleItemRename,
+  onItemValueLoad: handleItemValueLoad,
+}: {
+  readonly statValue: string | number | undefined;
+  readonly urlToFetch: string;
+  readonly pathToValue: string;
+  readonly statLabel: string;
+  readonly isDefault: boolean;
+  readonly formatter: (rawValue: any) => string;
+  readonly onClick: (() => void) | undefined;
+  readonly onRemove: (() => void) | undefined;
+  readonly onItemRename: ((newLabel: string) => void) | undefined;
+  readonly onItemValueLoad:
+    | ((value: number | string, itemLabel: string) => void)
+    | undefined;
+}): JSX.Element {
+  const [count] = useAsyncState<number | string | undefined>(
+    React.useCallback(
+      async () =>
+        statValue === undefined
+          ? throttledAjax<BackendStatsResult, string>(
+              'backendStats',
+              async () =>
+                ajax<BackendStatsResult>(urlToFetch, {
+                  method: 'GET',
+                  headers: {
+                    Accept: 'application/json',
+                  },
+                }).then(({ data }) => data),
+              urlToFetch
+            ).then((data) => {
+              const rawValue = data[pathToValue as keyof BackendStatsResult];
+              if (rawValue === undefined) return undefined;
+              return formatter(rawValue);
+            })
+          : Promise.resolve(undefined),
+      [statLabel, statValue, handleItemValueLoad, pathToValue, urlToFetch]
+    ),
+    false
+  );
+  if (count !== undefined && statValue === undefined) {
+    handleItemValueLoad?.(count, statLabel);
```

Comment:
this is dangerous and very bug prone (i.e, what if React calls your render function twice with the same props?

If you don't want things to be done twice, they should be done inside of useEffect